### ISSUE TITLE
Add hide functionality for tab-bar [Feature Request]

### ISSUE THREAD

**Issue by Robbendebiene** [Quicksaver/The-Fox--Only-Better#111]
Hey, I really like your addons, they make really powerful changes/improvements to the firefox UI.
Thanks for that :)

So for now i'm using Puzzle Toolbars and The-Fox-Only-Better, but even if I like the style of the hiding address bar, I would find it more useful to hide the tab-bar, because it is not that much needed by visiting a page than the address bar with it's menu, addons etc.

So I have no idea how difficult the implementation would be, but maybe you like the idea and find some time in the future to add this.

keep up the great work!

**Comment by Quicksaver** [comment]
Well, implementing it is difficult to say the least. But that's not the biggest problem that I can see with this idea.

How do you suggest this would work? The tabs can't be shown above the window, so they'd have to be moved below the navigation bar at least; this would make it so there would be less space to show tabs in, not to mention most drag and drop operations would be more difficult. You'd also lose loading/status/title indicators for background tabs.

**Comment by Quicksaver** [comment]
Plus, without a constant visual indicator, it would be easy to forget what and how many tabs you have open.

**Comment by Robbendebiene** [comment]
Okay, as I understand you would have to completely recreate the tab-bar, which would be very complex. The only thing I do not understand is, why there would be less space to show tabs.

I thought about something similar to the full screen mode, but only for the tab-bar. So the tab-bar is sliding in from the top and pushing down the address bar. In this case it might be better to change the address-bars layer so it lies over the page in contrast to the fullscreen mode which is changing the window size when the address and tab bar is moving down.

I could make a little jsfiddle if you want a better illustration for what i mean, but I think you still see a lot of problems there. I also wonder what will happen with the the minimize and close buttons? I would move them to the address-bar but I think this is not possible.

**Comment by Quicksaver** [comment]
> The only thing I do not understand is, why there would be less space to show tabs.

For the same reason that currently it shows the location bar / buttons in less space. Only the "Full width" style would make available the same amount of space. This is a more important point for the tabs than it is for the other toolbars.

The thing is, you wouldn't be able to hide both the tabs bar and the navigation bar, otherwise there would be nothing at the top to move the mouse into, to show them afterwards. So *if* there would be this feature, it would at least have to behave sort of like "hide navigation bar | hide tabs bar | hide other toolbars only". (To hide everything, you can just use fullscreen after all).

Side-note: I'm not overly fond of leaving the navigation toolbar at the top though. You'll lose the common window abilities, like dragging the window somewhere, double clicking it to maximize, etc. Only the tabs bar has this ability because only the tabs bar has "free space" to allow it. So I'd only even consider this if I was able to find a way to give the navigation bar those same powers.

> I also wonder what will happen with the the minimize and close buttons?

I've never really looked at them with this in mind, but I think that might be the easiest part even. :)

**Comment by Robbendebiene** [comment]
> For the same reason that currently it shows the location bar / buttons in less space.

Now I understand what you mean, but as you mentioned this would not be a problem with the "Full width" style. 

> The thing is, you wouldn't be able to hide both the tabs bar and the navigation bar, otherwise there would be nothing at the top to move the mouse into, to show them afterwards.

Yeah it wasn't my purpose to hide both bars, I just tried to explain my "idea" of how it should/could look :) Actually I would like these functions: "hide navigation bar | hide tabs bar | hide other toolbars only" 

> Side-note: I'm not overly fond of leaving the navigation toolbar at the top though. You'll lose the common window abilities, like dragging the window somewhere, double clicking it to maximize, etc.

You've got a point there, even if I'm not using these functions cause of a too huge amount of tabs, others surely do. You could make an extra button or a little space for that, where you can perform all these actions, but I know that this isn't a nice solution
Anyway I bet those people who need these functionalities wont use the "hide tab-bar" style :P

**Comment by Quicksaver** [comment]
> I bet those people who need these functionalities wont use the "hide tab-bar" style

Agreed. But if it's not done right, then no one will use it at all, including yourself I'm guessing. :)

By the way, I hope you understand that this won't be very high in my priorities list, as there's a lot of other things that I would like to add before this.

**Comment by Robbendebiene** [comment]
Yeah I'm totally fine with that. It would be very outrageous to demand this from you while you just work on all this stuff in your free time and gain almost nothing from it :)

Lastly i still have a little feature request, maybe I should make another thread for it, but I hope this won't be that difficult.
Can you add a shortcut to just show and hide the Slim Chrome, also for Mouse Buttons (like MouseButton4)?

**Comment by Quicksaver** [comment]
Do you mean use a mouse button as if it were a keyboard key? And if yes, do you mean to toggle Slim Chrome on and off or just to show it?

**Comment by Robbendebiene** [comment]
> Do you mean use a mouse button as if it were a keyboard key? 
Yes, instead of F6 or something like that i want to use an extra mouse button.

>  And if yes, do you mean to toggle Slim Chrome on and off or just to show it?
Just show it.

**Comment by simonbcn** [comment]
:+1:
